geoh5py/shared/merging/drape_model.py

```diff
@@ -139,32 +139,30 @@ def merge_data(
     ):
         super().merge_data(out_entity, input_entities)
 
-        # prepare an empty nan array
-        values: np.ndarray = np.empty(out_entity.n_cells)
-        values[:] = np.nan
+        ind_map = []
+        data_count = 0
+        ghost_count = 0
+        n_values = np.sum([input_entity.n_cells for input_entity in input_entities])
+        for input_entity in input_entities:
+            if input_entity.n_cells is None:
+                continue
+
+            n_cells = input_entity.n_cells
```

Comment:
Seems a lot of code lines . 
What about putting 

`ind_map.append([n_values + ghost_count, n_values + ghost_count + 1])`at the beginning of the loop.

initiate `ghost_count`to -2 or even better, use enumerate instead (ghost then `(i-1)*2`).

replace `np.hstack(ind_map)` by `np.hstack(ind_map[1:])`

saving 2 lines and an if condition. 

Sure it's not a big deal but made the code prettier.

Reply:
Maybe? Fine by me. Maybe better would be to do it on your PR once this is merged in.